temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
     int: DType.INT64,
 }
 
+DTYPE_MAPPING = {
+    np.float64: DType.FLOAT64,
+    np.float32: DType.FLOAT32,
+    np.int64: DType.INT64,
+    np.int32: DType.INT32,
+    np.str_: DType.STRING,
+    np.string_: DType.STRING,
+    np.bool_: DType.BOOLEAN,
+}
+DTYPE_REVERSE_MAPPING = {
+    DType.FLOAT64: np.float64,
+    DType.FLOAT32: np.float32,
+    DType.INT64: np.int64,
+    DType.INT32: np.int32,
+    DType.STRING: np.str_,
+    DType.BOOLEAN: np.bool_,
+}
+
+
+@dataclass
+class IndexData:
+    features: List[np.ndarray]
+    timestamps: np.ndarray
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray]],
+        timestamps: np.ndarray,
+    ) -> None:
+        if isinstance(features, np.ndarray):
+            features = [features]
+
+        self.features = features
+        self.timestamps = timestamps
+
+    def __len__(self) -> int:
+        return len(self.timestamps)
+
```

Comment:
I'd really like to have the `__getitem__` magic here (returning a tuple (timestamp, row_features)), and also an `.item()` method that only works when `IndexData` has 1 element and raises ValueError otherwise (like [torch](https://pytorch.org/docs/stable/generated/torch.Tensor.item.html)). I'd imagine that in many use cases, there's only 1 row per index, and it's clumsy to do `timestamp, features = event[0][0]`.
We can discuss this further anyway if it's not trivial.

Reply:
Just to clarify, I don't think this is important for the MVP, maybe it's one of those things that may come out from the UX testing. So feel free to stop reading this comment 😅 

So, in the general case, each `IndexData` actually contains a list of rows with `timestamp, features` pairs. So, for the general case we could access each row using `index_data[k]` instead of `index_data.timestamps[k]` + `index_data.features[k]`.
Aside of that, for the particular case where `index_data` only contains 1 row (i.e: the index is unique), there could be a shorthand method `index_data[0] -> index_data.item() ` which allows iteration over event rows like: `timestamp, features = event[p].item()`.

Reply:
I really like this idea. We can implement it in a future PR.